### PR TITLE
ifdef/remove deprecated methods

### PR DESCRIPTION
**Pull request by adjamhub**
The QSignalMapper::mapped is a trivial ifdef. The QModelIndex change needs deep review.

### CODE REVIEW

File: lxqt-config-session/autostartmodel.cpp [hunk 1]
```diff
@@ -150,7 +150,7 @@ bool AutoStartItemModel::removeRow(int row, const QModelIndex& parent)
 
     if (!mItemMap[item].removeLocal())
     {
-        QModelIndex index = parent.child(row, 0);
+        QModelIndex index = this->index(row, 0, parent);
```
Comment (tsujan) [inline lxqt-config-session/autostartmodel.cpp:153]: There's no need to `this->`.

Comment (adjamhub) [reply lxqt-config-session/autostartmodel.cpp:153]: Sure!
I put it there just because of the variable name :)

Comment (tsujan) [reply lxqt-config-session/autostartmodel.cpp:153]: Oh, yes! Please change the variable name to `indx`.

**Review by tsujan — APPROVED** [review]
GTM

**Pull request merged**